packages/react/src/date-picker/DatePicker.tsx

```diff
@@ -251,24 +266,25 @@ export const DatePicker = ({
 				value={inputValue}
 			/>
 			<CalendarProvider yearRange={yearRange}>
-				{/* We duplicate the Popover + Calendar as a workaround for a bug that scrolls the page to the top on initial open of the calandar - https://github.com/gpbl/react-day-picker/discussions/2059 */}
-				{hasCalendarOpened ? ( // If the calendar has opened at least once, we conditionally render the Popover + children to prevent the scroll-to-top bug
-					isCalendarOpen && (
-						<Popover {...popoverProps}>
-							<CalendarSingle {...calendarProps} />
-						</Popover>
-					)
-				) : (
-					// If the calendar has _not_ opened at least once, we conditionally render only the children of the Popover, i.e. the Calendar to prevent the UI jumping about everytime the calendar is opened
+				{isCalendarOpen && (
 					<Popover
 						{...popoverProps}
 						css={{ minHeight: '200px' }} // Using 200px as a safety buffer so that when opening the date picker for the first time and the input is at the bottom of the screen, it can't render the calendar almost hidden, e.g. 2px height.
 						visibility={isCalendarOpen ? 'visible' : 'hidden'}
 					>
-						{isCalendarOpen && <CalendarSingle {...calendarProps} />}
+						<CalendarSingle {...calendarProps} />
 					</Popover>
 				)}
 			</CalendarProvider>
 		</div>
 	);
 };
+
+// Attempts focus on target and returns true if successful
+function focusDay(queryTarget: string) {
```

Comment:
This is now created 4 times. Let's just do either 1 and import it everywhere, or 2 (legacy and next) and import in the range ones